lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp

```diff
@@ -0,0 +1,485 @@
+        if (!formatterResource)
+        {
+            return E_OUTOFMEMORY;
+        }
+
+        *resource = formatterResource;
+        return S_OK;
+    }
+
+    HRESULT SetNumberFormatSignificantDigits(IPlatformAgnosticResource *resource, const uint16 minSigDigits, const uint16 maxSigDigits)
+    {
+        auto *numFormatResource = reinterpret_cast<PlatformAgnosticIntlObject<icu::NumberFormat> *>(resource);
+        icu::NumberFormat *nf = numFormatResource->GetInstance();
+        // REVIEW (doilij): Without RTTI support, we can't do dynamic_cast,
+        // which is the preferred way to know whether it was indeed created as a DecimalFormat,
+        // which is not something the API will indicate to us directly (create returns a NumberFormat which may or may not be a DecimalFormat)
+        icu::DecimalFormat *df = reinterpret_cast<icu::DecimalFormat *>(nf);
+        df->setMinimumSignificantDigits(minSigDigits);
+        df->setMaximumSignificantDigits(maxSigDigits);
+        return S_OK;
+    }
+
+    HRESULT SetNumberFormatIntFracDigits(IPlatformAgnosticResource *resource, const uint16 minFracDigits, const uint16 maxFracDigits, const uint16 minIntDigits)
+    {
+        auto *numFormatResource = reinterpret_cast<PlatformAgnosticIntlObject<icu::NumberFormat> *>(resource);
+        icu::NumberFormat *nf = numFormatResource->GetInstance();
+        nf->setMinimumIntegerDigits(minIntDigits);
+        nf->setMinimumFractionDigits(minFracDigits);
+        nf->setMaximumFractionDigits(maxFracDigits);
+        return S_OK;
+    }
+
+    // We explicitly declare these specializations of FormatNumber so the compiler creates them
+    // because they will be used in another compilation unit,
+    // at which time we cannot generate code for specializations of this template.
+    template const char16 *FormatNumber<>(IPlatformAgnosticResource *formatter, const int32_t val, const uint16 formatterToUse, const uint16 currencyDisplay, const char16 *currencyCode);
+    template const char16 *FormatNumber<>(IPlatformAgnosticResource *formatter, const double val, const uint16 formatterToUse, const uint16 currencyDisplay, const char16 *currencyCode);
+
+    template <typename T>
+    const char16 *FormatNumber(IPlatformAgnosticResource *formatter, const T val, const uint16 formatterToUse, const uint16 currencyDisplay, const char16 *currencyCode)
+    {
+        icu::UnicodeString result;
+
+        auto *formatterHolder = reinterpret_cast<PlatformAgnosticIntlObject<icu::NumberFormat> *>(formatter);
+        if (!formatterHolder)
+        {
+            AssertOrFailFastMsg(formatterHolder, "Formatter did not hold an object of type `FinalizableIntlObject<icu::NumberFormat>`");
+            return nullptr;
+        }
+
+        icu::NumberFormat *numberFormatter = formatterHolder->GetInstance();
+
+        // __formatterToUse: 0 (default) is number, 1 is percent, 2 is currency
+        if (formatterToUse == 0 || formatterToUse == 1)
```

Comment:
Is `formatterToUse` only used within ChakraCore? I don't see a reason not to use an enum for it. 👍

Reply:
These numbers are also used on the JS side -- the effect was that it moved parsing String options to the JS side and made it a simple comparison on the native side as an int.

I can mirror an enum (or constants since JS doesn't really have enums) over to the JS as well so that we stop leaving these magic constants with comments everywhere :)